templates/base/server/db/auth/dao.py

```
@@ -0,0 +1,165 @@
+        return devices.all()
+
+    async def insert_device(self, device_data: DeviceData) -> str:
+        try:
+            new_device = Device(**device_data.model_dump())
+            self.session.add(new_device)
+            await self.session.commit()
+            await self.session.refresh(new_device)
+            return new_device.id
+        except Exception:
+            await self.session.rollback()
+            raise DeviceNotCreatedException()
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)